index.bs

```diff
@@ -52,10 +70,196 @@ dictionary StorageBucketOptions {
 </xmp>
 
 <h3 id="storage-bucket-open">Creating a bucket</h3>
+
+<div algorithm>
+
+The <dfn method for="StorageBucketManager">open(|name|, |options|)</dfn> method steps are:
+
+1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+    
+1. Let |shelf| be the result of running [=obtain a local storage shelf=] given |environment|.
+
+1. If |shelf| is failure, then return [=a promise rejected with=] a {{TypeError}}.
+
+1. If the result of [=validate a bucket name=] with |name| is failure,  then return [=a promise rejected with=] a {{TypeError}}.
+
+1. Let |r| be the result of running [=open a bucket=] with |shelf|, |name|, and |options|.
+
+1. If |r| is failure, then return [=a promise rejected with=] a {{TypeError}}.
+
+1. Return [=a promise resolved with=] with |r|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>open a bucket</dfn> for a |shelf| given a bucket |name| and optional |options|, run the following steps:
+
+1. Let |expires| be undefined.
+    
+1. If |options|["{{StorageBucketOptions/expires}}"] exists, then:
+    
+    1. Let |expires| be |options|["{{StorageBucketOptions/expires}}"].
+
+    1. If |expires| is less than or equal to the [=current wall time=], then return failure.
```

Comment:
So `|expires|` is a DOMHighResTimestamp and `[=current wall time=]` is a `[=moment=]`.  It seems we need some kind of conversion of the DOMHighResTimestamp to a moment.  Probably by adding the value to the unix time epoch?

@noamr Is there an example of this we can look at?  Can you review the usage of time in this PR?  It feels like we've gone back and forth a bit.  It would be good to get an authoritative review.

Reply:
Yep, we should compare in a type-correct way, using wording like `If the [=relevant settings object=]'s [=environment settings object/current wall time=] is less than |expires| milliseconds after the [=Unix epoch=]`. Or, like I mentioned above, have you considered using `maxAge` instead of `expires` so that you'd be comparing against 0 instead of the current time?

Reply:
SG, thank you!